src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposableFieldsShouldBeDisposedTests.cs

```diff
@@ -651,6 +651,186 @@ End Function
             }.RunAsync();
         }
 
+        [Fact, WorkItem(6075, "https://github.com/dotnet/roslyn-analyzers/issues/6075")]
+        public async Task AsyncDisposableDisposedInExplicitAsyncDisposable_Disposed_NoDiagnosticAsync()
+        {
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithAsyncInterfaces,
+                TestCode = @"
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+class FileStream2 : IAsyncDisposable
+{
+    public ValueTask DisposeAsync() => default;
+}
+
+public sealed class Test : IAsyncDisposable, IDisposable
+{
+    private readonly HttpClient client;
+    private readonly FileStream2 stream;
+
+    public Test()
+    {
+        client = new HttpClient();
+        stream = new FileStream2();
+    }
+
+    public void Dispose()
+    {
+        client.Dispose();
+    }
+
+    async ValueTask IAsyncDisposable.DisposeAsync()
+    {
+        await stream.DisposeAsync();
+    }
+}
+"
+            }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithAsyncInterfaces,
+                TestCode = @"
+Imports System
+Imports System.IO
+Imports System.Net.Http
+Imports System.Threading.Tasks
+
+class FileStream2 
+	implements IAsyncDisposable
+	public function DisposeAsync() as ValueTask implements IAsyncDisposable.DisposeAsync
+		return nothing
+	end function
+end class
+
+public class Test 
+	implements IAsyncDisposable, IDisposable
+	
+    private readonly client as HttpClient
+    private readonly stream as FileStream2
+
+	public sub new()
+        client = new HttpClient
+        stream = new FileStream2
+	end sub
+
+	public sub Dispose() implements IDisposable.Dispose
+        client.Dispose()
+	end sub
+
+	function DisposeAsync() as ValueTask implements IAsyncDisposable.DisposeAsync
+		return stream.DisposeAsync()
+	end function
+end class
+"
+            }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithAsyncInterfaces,
+                TestCode = @"
+Imports System
+Imports System.IO
+Imports System.Net.Http
+Imports System.Threading.Tasks
+
+class FileStream2 
+	implements IAsyncDisposable
+	public function DisposeAsync() as ValueTask implements IAsyncDisposable.DisposeAsync
+		return nothing
+	end function
+end class
+
+public class Test 
+	implements IAsyncDisposable, IDisposable
+	
+    private readonly client as HttpClient
+    private readonly stream as FileStream2
+
+	public sub new()
+        client = new HttpClient
+        stream = new FileStream2
+	end sub
+
+	public sub Dispose() implements IDisposable.Dispose
+        client.Dispose()
+	end sub
+
+    rem arbitrary implementation name
```

Comment:
Oh its been ages since I saw use of `rem` instead of the `'` for VB comments :-)!! Good old days...

Reply:
I had to google how to do a struct in VB, I haven't used it since like VB6 lol